storey/flow.py

```diff
@@ -1511,16 +1513,30 @@ async def run_async(self, body: Any, path: str) -> Any:
         return body
 
     def _run(self, body: Any, path: str) -> Any:
+        timestamp = datetime.datetime.now(tz=datetime.timezone.utc)
         start = time.monotonic()
-        body = self.run(body, path)
+        try:
+            body = self.run(body, path)
+        except Exception as e:
+            if not self._raise_exception:
+                body = {"error": f"{type(e)}: {e}"}
+            else:
+                raise e
         end = time.monotonic()
-        return _ParallelExecutionRunnableResult(self.name, body, end - start)
+        return _ParallelExecutionRunnableResult(self.name, body, end - start, timestamp)
 
     async def _async_run(self, body: Any, path: str) -> Any:
+        timestamp = datetime.datetime.now(tz=datetime.timezone.utc)
         start = time.monotonic()
-        body = await self.run_async(body, path)
+        try:
+            body = await self.run_async(body, path)
+        except Exception as e:
+            if not self._raise_exception:
+                body = {"error": f"{type(e)}: {e}"}
+            else:
+                raise e
```

Comment:
Very minor thing but I'd reverse the condition (elsewhere as well).
```suggestion
            if self._raise_exception:
                raise e
            else:
                body = {"error": f"{type(e)}: {e}"}
```